feat: add modern.js addon for storybook rsbuild (#294)
rspack-contrib/storybook-rsbuild: website/docs/guide/integrations/modernjs.mdx

We can use the environment variable to check if the addon is used in Storybook.

```json title="package.json"
{
  "scripts": {
    "storybook": "STORYBOOK=true storybook dev -p 6006"
  }
}
```

```ts title="modern.config.ts"
import { appTools, defineConfig } from '@modern-js/app-tools'
import { moduleFederationPlugin } from '@module-federation/modern-js'

// https://modernjs.dev/en/configure/app/usage
export default defineConfig({
  runtime: {
    router: true,
  },
  plugins: [
    appTools({
      bundler: 'rspack',
    }),
    ...(process.env.STORYBOOK ? [] : [moduleFederationPlugin()]),
  ],
})
```

## Limitations

Modern.js has its own plugin system, which might internally handle the configurations of Modern.js and Rsbuild. 

Modern.js also processes user configurations internally, performing complex transformations before passing them to Rsbuild.

:::tip
Therefore, the addon has some limitations in Modern.js.
:::

### 🎗 Supported Usages

- Modern.js Configuration files
- Modern.js plugins to modify the configuration
- Rsbuild plugins

### 😪 Unsupported Usages

> - Webpack Bundler cannot be used, only Rspack is supported.

Reviewer: ```suggestion
- webpack cannot be used, only Rspack is supported.
```

> - SSR Configuration will not work in Storybook

Reviewer: ```suggestion
- SSR configuration will not work in Storybook
```

- Runtime capabilities, such as `<Link>` navigation

> - [Modern.js plugins](https://modernjs.dev/plugin/cli-plugins/api.html) only run the `config` and `resolveConfig` hooks

Reviewer: ```suggestion
- [Modern.js plugins](https://modernjs.dev/plugin/cli-plugins/api.html) will only call the `config` and `resolveConfig` hooks
```

### ☎️ Contact Us

> We are now only supporting the Storybook addon for Modern.js, not the Storybook Framework. Which means some features of Modern.js are not supported in Storybook.

Reviewer: ```suggestion
We now support Storybook through a Storybook addon, not a Storybook framework. As a result, some features of Modern.js are unavailable in Storybook.
```
